library/Faktory/Worker.hs

```diff
@@ -90,8 +91,12 @@ processorLoop client settings workerSettings f = do
   let
     namespace = connectionInfoNamespace $ settingsConnection settings
     processAndAck job = do
-      f job
-      ackJob client job
+      mResult <- timeout (getJobTTL job) $ do
+        f job
+        ackJob client job
```

Comment:
Should `timeout` wrap only `f` instead of the `f` and `ACK`?

Reply:
Yeah, I meant something like:

```hs
mResult <- timeout _ $ j job

case mResult of
  Nothing -> ...
  Just () -> do
    ackJob client job
```

This is super unlikely to matter, but it seems complicated to think about the timing between three events:

- `timeout` interrupts
- `ACK` sent to Faktory
- Faktory releases reservation

With a `timeout` exactly equal to what Faktory is working with, and `ACK` being (I think) blocking I/O, it could get interesting.

Reply:
Fair